File: daraja/src/main/java/com/twigafoods/daraja/model/LNMExpress.java
package com.twigafoods.daraja.model;

import com.twigafoods.daraja.util.Settings;

public class LNMExpress {
    private String BusinessShortCode;
    private String Password;
    private String Timestamp;
    private String TransactionType;
    private String Amount;
    private String PartyA;
    private String PartyB;
    private String PhoneNumber;
    private String CallBackURL;
    private String AccountReference;
    private String TransactionDesc;

    public LNMExpress(String businessShortCode, String transactionType, String amount, String partyA, String partyB, String phoneNumber, String callBackURL, String accountReference, String transactionDesc) {
        BusinessShortCode = businessShortCode;
        TransactionType = transactionType;
        Amount = amount;
        PartyA = partyA;
        PartyB = partyB;
        PhoneNumber = phoneNumber;
        CallBackURL = callBackURL;
        AccountReference = accountReference;
        TransactionDesc = transactionDesc;
    }

    public String getBusinessShortCode() {
        return BusinessShortCode;
    }

    public void setBusinessShortCode(String businessShortCode) {
        BusinessShortCode = businessShortCode;
    }

    public String getPassword() {
        return Password;
    }

    public void setPassword(String password) {
        Password = Settings.generatePassword(this.BusinessShortCode, "", this.Timestamp);
    }

    public String getTimestamp() {
        return Timestamp;
    }

    public void setTimestamp(String timestamp) {
        Timestamp = Settings.generateTimestamp();
    }

    public String getTransactionType() {
        return TransactionType;
    }

    public void setTransactionType(String transactionType) {
        TransactionType = transactionType;
    }

    public String getAmount() {
        return Amount;
    }

    public void setAmount(String amount) {
        Amount = amount;
    }

    public String getPartyA() {
        return PartyA;
    }

    public void setPartyA(String partyA) {
        PartyA = partyA;
    }

    public String getPartyB() {
        return PartyB;
    }

    public void setPartyB(String partyB) {
        PartyB = partyB;
    }

    public String getPhoneNumber() {
        return PhoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        PhoneNumber = phoneNumber;
    }

    public String getCallBackURL() {
        return CallBackURL;
    }

    public void setCallBackURL(String callBackURL) {
        CallBackURL = callBackURL;
    }

    public String getAccountReference() {
        return AccountReference;
    }

    public void setAccountReference(String accountReference) {
        AccountReference = accountReference;
    }

    public String getTransactionDesc() {
        return TransactionDesc;
    }

    public void setTransactionDesc(String transactionDesc) {
        TransactionDesc = transactionDesc;
    }
}

File: daraja/src/main/java/com/androidstudy/daraja/model/LNMExpress.kt
package com.androidstudy.daraja.model

import com.androidstudy.daraja.constants.TransactionType

class LNMExpress {

    private var BusinessShortCode: String
    private lateinit var PassKey: String
    private lateinit var Password: String
    private lateinit var Timestamp: String
    private lateinit var Type: TransactionType
    private var Amount: String
    private lateinit var TransactionType: String
    private var PartyA: String
    private var PartyB: String
    private var PhoneNumber: String
    private var CallBackURL: String
    private var AccountReference: String
    private var TransactionDesc: String

    constructor(businessShortCode: String, password: String, timestamp: String, amount: String, transactionType: String, partyA: String, partyB: String, phoneNumber: String, callBackURL: String, accountReference: String, transactionDesc: String) {
        BusinessShortCode = businessShortCode
        Password = password
        Timestamp = timestamp
        TransactionType = transactionType
        Amount = amount
        PartyA = partyA
        PartyB = partyB
        PhoneNumber = phoneNumber
        CallBackURL = callBackURL
        AccountReference = accountReference
        TransactionDesc = transactionDesc
    }

    constructor(businessShortCode: String, passKey: String, transactionType: TransactionType, amount: String, partyA: String, partyB: String, phoneNumber: String, callBackURL: String, accountReference: String, transactionDesc: String) {
        BusinessShortCode = businessShortCode
        PassKey = passKey
        Type = transactionType
        Amount = amount
        PartyA = partyA
        PartyB = partyB
        PhoneNumber = phoneNumber
        CallBackURL = callBackURL
        AccountReference = accountReference
        TransactionDesc = transactionDesc
    }

}